velox/exec/TableWriter.h

```diff
@@ -113,25 +113,14 @@ class TableWriter : public Operator {
 
   void addInput(RowVectorPtr input) override;
 
-  void noMoreInput() override {
-    Operator::noMoreInput();
-    if (aggregation_ != nullptr) {
-      aggregation_->noMoreInput();
-    }
-    close();
-  }
+  void noMoreInput() override;
 
   virtual bool needsInput() const override {
     return true;
   }
 
   void close() override {
-    if (!closed_) {
-      if (dataSink_ != nullptr) {
-        dataSink_->close();
-      }
-      closed_ = true;
-    }
+    abortDataSink();
```

Comment:
This is confusing. Should we check if !closed_ here and then call abortDataSink with a comment explaining that we are aborting because there was some failure?